src/libraries/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HashAlgorithm.cs

```diff
@@ -99,24 +99,8 @@ public byte[] ComputeHash(Stream inputStream)
             if (_disposed)
                 throw new ObjectDisposedException(null);
 
-            // Use ArrayPool.Shared instead of CryptoPool because the array is passed out.
-            byte[] buffer = ArrayPool<byte>.Shared.Rent(4096);
-
-            int bytesRead;
-            int clearLimit = 0;
-
-            while ((bytesRead = inputStream.Read(buffer, 0, buffer.Length)) > 0)
-            {
-                if (bytesRead > clearLimit)
-                {
-                    clearLimit = bytesRead;
-                }
-
-                HashCore(buffer, 0, bytesRead);
-            }
+            inputStream.CopyTo((span, _) => HashCore(span), null, 4096);
```

Comment:
You should pass `this` in as state and use that to invoke HashCore to avoid the delegate allocation required to capture `this`.